FormalSystems/Chomsky/Regular/TotalDFA.lean

```diff
@@ -32,3 +32,96 @@ theorem in_language_iff_del_star_final
   w ∈ M.GeneratedLanguage ↔ M.del_star' (M.q₀, w) ∈ M.F := by
   rw [DFA.in_language_iff_del_star_final, total_del_star_eq]
   simp
+
+instance [i: Fintype t]: Fintype (Option t) where
+  elems :=
+    (i.elems.map ⟨ fun x => some x, fun _ _ => by simp ⟩).cons
+      none (by simp)
+  complete := fun x => by cases x <;> simp; apply Fintype.complete
```

Comment:
Since we are using mathlib anyway, is there a reason not to use [this](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Data/Fintype/Option.html#instFintypeOption)?

Reply:
Haha, I did not find this one, and now I know why: because I didn't import it.